From 26a9eb7b44b05d72111ed8ac0ab4a7f3d9a2141a Mon Sep 17 00:00:00 2001
From: Todd Short <tshort@akamai.com>
Date: Fri, 30 Aug 2019 09:15:31 -0400
Subject: [PATCH 12/43] QUIC: Add OPENSSL_NO_QUIC wrapper

---
 ssl/ssl_local.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ssl/ssl_local.h b/ssl/ssl_local.h
index fdce1f1c5c..c7b64d4c06 100644
--- a/ssl/ssl_local.h
+++ b/ssl/ssl_local.h
@@ -1504,9 +1504,11 @@ struct ssl_st {
     unsigned char handshake_traffic_hash[EVP_MAX_MD_SIZE];
     unsigned char client_app_traffic_secret[EVP_MAX_MD_SIZE];
     unsigned char server_app_traffic_secret[EVP_MAX_MD_SIZE];
+# ifndef OPENSSL_NO_QUIC
     unsigned char client_hand_traffic_secret[EVP_MAX_MD_SIZE];
     unsigned char server_hand_traffic_secret[EVP_MAX_MD_SIZE];
     unsigned char client_early_traffic_secret[EVP_MAX_MD_SIZE];
+# endif
     unsigned char exporter_master_secret[EVP_MAX_MD_SIZE];
     unsigned char early_exporter_master_secret[EVP_MAX_MD_SIZE];
     EVP_CIPHER_CTX *enc_read_ctx; /* cryptographic state */
-- 
2.47.3

